### PR TITLE
refactor: removed async module dependency

### PR DESCRIPTION
**Pull request by vijay-qlogic**
Fixes #[2883](https://github.com/googleapis/google-cloud-node/issues/2883) partially for nodejs-proto-files (it's a good idea to open an issue first for discussion)

- [ ] Tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: tools/prepublish.ts [hunk 1]
```diff
@@ -37,44 +37,36 @@ const extract = (input, opts, callback) => {
...
+async function main() {
+  await require('child_process').exec('rm -rf google');
 
-      (next) => {
-        extract(
-            'https://github.com/googleapis/googleapis/archive/master.zip', {
-              strip: 1,
-            },
-            next);
+  await extract(
```
Comment (JustinBeckwith) [inline tools/prepublish.ts]: Are you sure that `extract` returns a promise?  It looks like you're just passing an empty closure as the callback.  You may need to use `util.promisify` here.

File: tools/prepublish.ts [hunk 2]
```diff
@@ -37,44 +37,36 @@ const extract = (input, opts, callback) => {
...
+          return (
+              file.parent.indexOf('protobuf-master') === 0 &&
+              file.parent.indexOf('protobuf-master/src/') === 0 &&
+              file.parent.indexOf('/compiler') === -1 &&
+              file.parent.indexOf('/internal') === -1 &&
+              file.filename.indexOf('unittest') === -1 &&
+              file.filename.indexOf('test') === -1);
+        },
       },
+      () => {});
```
Comment (JustinBeckwith) [inline tools/prepublish.ts]: Same feedback, this probably doesn't do what we want

File: tools/prepublish.ts [hunk 3]
```diff
@@ -37,44 +37,36 @@ const extract = (input, opts, callback) => {
       });
 };
 
-async.series(
-    [
-      (next) => {
-        require('child_process').exec('rm -rf google', next);
-      },
+async function main() {
+  await require('child_process').exec('rm -rf google');
```
Comment (JustinBeckwith) [inline tools/prepublish.ts]: Pretty sure this doesn't return a promise.  You will need to use `util.promisify` to wrap any of these

File: package.json [hunk 4]
```diff
@@ -37,6 +37,7 @@
     "pretest": "npm run prepare && npm run compile"
   },
   "dependencies": {
+    "@google-cloud/promisify": "^0.3.1",
```
Comment (JustinBeckwith) [inline package.json]: You don't need this :) The `util` module is baked into nodejs, and `util.promisify` works just fine.

File: tools/prepublish.ts [hunk 5]
```diff
@@ -4,8 +4,7 @@
  * Distributed under MIT license.
  * See file LICENSE for detail or copy at https://opensource.org/licenses/MIT
  */
-
-import * as async from 'async';
+import {promisify} from '@google-cloud/promisify';
```
Comment (JustinBeckwith) [inline tools/prepublish.ts]: Just `import {promisify} from 'util'` will do just fine here.

File: package.json (hunk 4 above)
Comment (vijay-qlogic) [reply package.json]: @JustinBeckwith I tried that, but it's failing in node-6 build, for .ts

File: tools/prepublish.ts (hunk 5 above)
Comment (vijay-qlogic) [reply tools/prepublish.ts]: @JustinBeckwith  Done, but it's failing for node6

Comment (JustinBeckwith) [reply tools/prepublish.ts]: 🤦‍♂️  I'm sorry.  I didn't realize the unit tests would be covering this directory for some reason.  Feel free to use the previous code.  Again, apologies for making you change it back.

**Pull request merged**